repos/system_upgrade/el8toel9/actors/nischeck/tests/test_nischeck.py

```diff
@@ -0,0 +1,83 @@
+import pytest
+
+from leapp import reporting
+from leapp.libraries.actor import nischeck
+from leapp.libraries.common.testutils import create_report_mocked, CurrentActorMocked
+from leapp.libraries.stdlib import api
+from leapp.models import InstalledRedHatSignedRPM, NISConfig, RPM
+
+
+def _generate_rpm(name, version, release, epoch):
+    """
+    Generate new RPM model.
+
+    Parameters:
+        name (str): rpm name
+        version (str): rpm version
+        release (str): rpm release
+        epoch (str): rpm epoch
+
+    Returns:
+        rpm  (RPM): new RPM object with name parameter set
+    """
+    return RPM(name=name,
+               version=version,
+               release=release,
+               epoch=epoch,
+               pgpsig='RSA/SHA256, Mon 01 Jan 1970 00:00:00 AM -03, Key ID 199e2f91fd431d51',
+               packager='Red Hat, Inc. <http://bugzilla.redhat.com/bugzilla>',
+               arch='noarch')
+
+
+@pytest.mark.parametrize('pkgs_installed', [
+    ('ypserv',),
+    ('ypbind',),
+    ('ypserv-not-ypbind',),
+    ('ypbind', 'ypserv'),
+])
+@pytest.mark.parametrize('pkgs_configured', [
+    (),
+    ('ypbind',),
+    ('ypserv',),
+    ('ypbind', 'ypserv'),
+])
+def test_actor_nis(monkeypatch, pkgs_installed, pkgs_configured):
+    """
+    Parametrized helper function for test_actor_* functions.
+
+    First generate list of RPM models based on set arguments. Then, run
+    the actor feeded with our RPM list and mocked functions. Finally, assert
+    Reports according to set arguments.
+
+    Parameters:
+        pkgs_installed  (touple): installed pkgs
+        fill_conf_file  (bool): not default ypbind config file
+        fill_ypserv_dir  (bool): not default ypserv dir content
+    """
+
+    # Generate few standard packages
+    rpms = [_generate_rpm('rclone', '2.3', '3', '1'),
+            _generate_rpm('gdb', '1.0.1', '1', '2')]
+
+    # Generate packages from 'pkgs_installed'
+    for pkg_name in pkgs_installed:
+        rpms += [_generate_rpm(pkg_name, '2.0', '3', '1')]
+
+    # Generate NIS facts
+    nis_facts = NISConfig(nis_not_default_conf=pkgs_configured)
+
+    curr_actor_mocked = CurrentActorMocked(msgs=[InstalledRedHatSignedRPM(items=rpms), nis_facts])
+    monkeypatch.setattr(api, 'current_actor', curr_actor_mocked)
+    monkeypatch.setattr(reporting, "create_report", create_report_mocked())
+
+    # Executed actor feeded with out fake msgs
+    nischeck.report_nis()
+
+    # Check for configured NIS package
+    for pkg in pkgs_configured:
+        if pkg in pkgs_installed:
+            assert reporting.create_report.called == 1
+            break
+    else:
+        # Assert for no NIS installed and configured package
+        assert not reporting.create_report.called
```

Comment:
TBH this is quite kind confusing to read. 

If I understand the intention is that is only called once if a pkg is installed and configured and not called if not.

```suggestion
    # Check if there any relevant packages installed with a modified configuration
    if set(pkgs_configured) & set(pkgs_installed)):
        # Ensure that it is reported only once
        assert reporting.create_report.called == 1
    else:
        # Otherwise ensure that no report is generated
        assert not reporting.create_report.called    
```

Reply:
That's different logic proposed by you. We are 'faking' two variables here with fixtures. `pkgs_configured` and `pkgs_installed`. pytest will test all possible scenarios so we will get weird combination like: installed ypserv but configured ypbind.

I didn't make special condition for that in test code because there is no need. This is not possible scenario and we don't need to cover it.

What might be a bit confusing is that we start iteration with `pkgs_configured` and not `pkgs_installed`. It actually does not matter but I changed it and added comments.